### ISSUE TITLE
JS error after URL verification

### ISSUE THREAD

**Issue by layoutd** [woocommerce/pinterest-for-woocommerce#899]
### Describe the bug:
After I verify my URL, I'm dumped onto a blank page:
![Z3rlgW.gif](https://github.com/woocommerce/pinterest-for-woocommerce/assets/228780/28362d0b-94cb-4735-8369-19669ed4e396)
   - The error is `TypeError: Cannot read properties of undefined (reading 'creditsGiven')`
   ![image](https://github.com/woocommerce/pinterest-for-woocommerce/assets/228780/e0c05167-831a-4c28-8425-27b2b9a38bc5)
   `("As a new Pinterest customer, you can get %1$s in free ad credits when you successfully set up Pinterest for WooCommerce and spend %2$s on Pinterest Ads. <a>Pinterest Terms and conditions</a> apply.", "pinterest-for-woocommerce"), a.creditsGiven, a.spendRequire)`

Refreshing shows a verified site and I can continue with the onboarding.
![EF5PWY.gif](https://github.com/woocommerce/pinterest-for-woocommerce/assets/228780/1cddfa1f-b87e-4698-9db4-ef23de389e3c)

### Steps to reproduce:
1. Do onboarding. WC 8.5 or 8.6. Store not fully set up. I was using `ddev share` to get an ngrok site.
2. Verify a domain.

### Expected behavior:
Continue onboarding flow

### Actual behavior:
Blank screen until referesh

**Comment by layoutd** [comment]
After clicking verify, the async call goes to `/wp-json/pinterest/v1/domain_verification?_locale=user`.
The response is
```json
{
  "status": "success",
  "website": "1111-111-11-111-1111.ngrok-free.app",
  "account_data": {
    "full_name": "My Account",
    "is_partner": true,
    "image_medium_url": "https://i.pinimg.com/originals/31/68/ff/3168ff7903ec0ede7b77c3ece227fa78.gif",
    "is_any_website_verified": true,
    "username": "woocom",
    "verified_user_websites": [
      "url1.com",
      "url2.com",
      "url3.com",
      "url4.com",
      "url5.com",
      "1111-111-11-111-1111.ngrok-free.app"
    ],
    "id": "811111111111111116",
    "is_billing_setup": false,
    "coupon_redeem_info": {
      "redeem_status": false
    }
  }
}
```
I suspect this has to do with having other domains verified already.

**Comment by layoutd** [comment]
Maybe not, it looks like that call doesn't include
```json
            "currency_credit_info": {
                "spendRequire": "15\u00a0\u20ac",
                "creditsGiven": "131\u00a0\u20ac"
            }
```
Which IS included in the response from `/wp-json/pinterest/v1/settings?_locale=user`

**Comment by sukafia** [comment]
I've been tracking a similar issue see: p1704234203000119/1697237207.778199-slack-C04QQLX04MC

**Comment by layoutd** [comment]
This is resolved in this PR pending review: #876

**Comment by layoutd** [comment]
Closed with #876
